README.md
# TheSaaS

TheSaaS is a responsive, professional, and multipurpose SaaS, Software, Startup and WebApp landing template powered by Bootstrap 4. TheSaaS is a powerful and super flexible tool, which suits best for any kind of landing pages. TheSaaS is definitely a great kick starter for your web project.

TheSaaS design is harmonious, clean and user friendly. Even though the template has a lot of content, it doesn’t looks messy and all files and code are well structured, commented and divided. TheSaaS has a huge collection of plugins and UI components and works seamlessly on all major web browsers, tablets and phones.

## What's included

Within the download package you'll find the following directories:

```
thesaas/
├── starter/
├── theme/
└── v1.5/
```

- `starter`: It includes starter templates to get started with. This is the place you should choose a starter, copy a directory and start your project with. There's a complete documentation inside a readme.html file for each starter template.
- `theme`: This directory contains all the files that you see in the online demo. You should use these files to copy codes or files from and develop your own project. Also, it's a documentation and reference for our components.
- `v1.5`: The previous major release of TheSaaS which is included for thousands od customers who has developed their project based on the v1.x.

## Versioning

For transparency into our release cycle and in striving to maintain backward compatibility, TheSaaS is maintained under [the Semantic Versioning guidelines](http://semver.org/). Sometimes we screw up, but we'll adhere to those rules whenever possible.

See CHANGELOG.md for list of changes in each release.


## Upgrade from TheSaaS v1.x to v2.x

This is the second major release of TheSaaS which is based on the stable release of Bootstrap 4. The previous version (v1.x) is based on the Bootstrap 4 alpha-6. As the version number implies, this release contains a lot of breaking changes due to breaking changes from Bootstrap 4. So there's not any guide on how to upgrade from TheSaaS v1.x to TheSaaS v2.x. Since the v1.x is already stable enough, you don't have to upgrade your project at all—unless you have enough time to recreate it using the current release of TheSaaS. But if you're starting new project, TheSaaS v2.x is the right choice.

## Creator

**TheThemeio**: <http://thetheme.io>

## Copyright and license

This product licensed under the [Themeforest Standard Licenses](https://themeforest.net/licenses/standard). If your end product including the item is going to be free to the end user then a [Regular License](https://themeforest.net/licenses/terms/regular) is what you need. An [Extended License](https://themeforest.net/licenses/terms/extended) is required if the end user must pay to use the end product.


CHANGELOG.md
# Change Log
All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](http://keepachangelog.com/) 
and this project adheres to [Semantic Versioning](http://semver.org/).


## [2.1.5] - 2019-01-14
- Update : NPM packages
- Update : Plugins (Bootstrap, Poppers, Granim, AOS Jarallax, Shufflejs, SmoothScroll)
- Update : Demo names and some of the images has changed
- Update : Some block movement between block pages and demo pages
- Update : Minor improvements in styles and scripts
- Update : reCAPTCHA to disable wrapper form from submission using Javascript if data-enable option presents.
- Fix    : Coming soon page count down was done.


## [2.1.4] - 2018-10-01
- Add    : onSuccess and onError callbacks to contact forms (See block #1)
- Update : NPM packages
- Fix    : Sticky navbar without background color
- Fix    : Switch control wasn't firing change event


## [2.1.3] - 2018-09-01
- Update : Some of the plugins and package.json
- Fix    : Wrong Img-Height when using 'img-fluid' inside a card in IE11
- Fix    : Custom checkbox and custom radio wasn't working on iOS devices
- Fix    : Error in "gulp dist" task, due to new version of webpack stream


## [2.1.2] - 2018-07-26
- Update : To Bootstrap v4.1.3


## [2.1.1] - 2018-07-23
- Add    : Layout #8 to /uikit/navbar with user avatar
- Add    : Ability to set pricing table's button link using data-bind-href
- Update : To Bootstrap v4.1.2
- Update : /docs/faq.html page to have scrollspy
- Fix    : Fullscreen video on IE11 using ObjectFitPolyFill
- Fix    : Logo scale issue on IE11


## [2.1.0] - 2018-06-18
- Add    : New page /demo/wordpress
- Add    : New page /demo/themeforest
- Add    : New page /demo/dribbble
- Add    : New page /block/content
- Add    : New page /block/download
- Add    : New page /block/explore
- Add    : New page /block/gallery
- Add    : New page /block/process
- Add    : New page /block/service
- Add    : New page /uikit/sidebar
- Add    : New page /blog/post-2
- Add    : New page /docs/index
- Add    : New page /docs/articles
- Add    : New page /docs/faq
- Add    : New page /docs/kb-*
- Add    : New page /page/about-2
- Add    : New page /page/about-3
- Add    : New page /page/contact-2
- Add    : New page /page/contact-3
- Add    : New page /page/service-1
- Add    : New page /page/service-2
- Add    : New /block/contact/ #6, #7, #8, #9
- Add    : New /block/cta/ #5, #6
- Add    : New /block/feature-text/ #9
- Add    : New /block/portfolio/ #7
- Update : Renamed /page/about to /page/about-1
- Update : Renamed /page/contact to /page/contact-1
- Update : Moved /block/cta #5, #6 to /block/download
- Update : Moved /block/feature-text #9 to /block/explore
- Fix    : Deprication warning from LibSass regards importing css file


## [2.0.4] - 2018-06-04
- Add    : /uikit/process with Timeline and Step components
- Add    : /block/feature #15
- Add    : /block/subscribe #6
- Add    : /block/testimonial #8
- Add    : /block/video #6
- Add    : New accordion variations
- Update : Improvements to the navbar for mobile devices
- Update : Mega menu items are in alphabetical order now
- Update : Replace some of the images
- Update : Some minor changes and enhancements


## [2.0.3] - 2018-05-26
- Update : Plugins (Bootstrap, Popper, Lity, Jarallax, etc.)
- Update : Close the navbar on mobile if user clicks on an anchor link
- Update : Some improvement in smart navbars types
- Update : Improvement in smooth scroll to an element
- Update : Minor improvements in CSS and JS code
- Fix    : Lots of improvement in mobile responsive behavior
- Fix    : Border radius of input-groups
- Fix    : Shaking problem for displaying a video in lity lightbox


## [2.0.2] - 2018-04-23
- Add    : Once popups to display a popup only for one time. See /uikit/popup.html
- Add    : .modal-long class to have modal with long content
- Add    : .navbar-offset-0 class to remove top offset from navbar
- Update : Check the expiration of reCAPTCHA to disable submit button
- Fix    : Shuffle had js error
- Fix    : Contact form didn't report error
- Fix    : Misspelling in data-video of /uikit/section.html which prevent the YouTube and Vimeo video to play
- Fix    : Few other minor bugs


## [2.0.1] - 2018-03-19
- Add    : Two starters with full documentation; basic and expert
- Add    : Form validation support and samples
- Add    : Centered reCAPTCHA and ability to enable a submit button after pass
- Add    : New pricing table sample to /uikit/pricing.html
- Add    : Gallery component; see /uikit/gallery.html
- Add    : Support multiple locations on Google map
- Add    : In page navigation. See sample on /uikit/nav.html
- Update : Include a top offset for the navbar
- Fix    : Responsive issue in footer of the main demo
- Fix    : Slider pagination problem on shop index
- Fix    : Supporting navbar-stick-(dark|light) for data-navbar="fixed"
- Fix    : Few minor bugs in style


## [2.0.0] - 2018-03-01
- Initial release!
- It has completely rewritten 



starter/expert/gulpfile.js
'use strict';

var gulp = require('gulp');
var webpack = require('webpack-stream');
var webpackUglifyJs = require('uglifyjs-webpack-plugin')
var sass = require('gulp-sass');
var tildeImporter = require('node-sass-tilde-importer');
var browserSync = require('browser-sync');
var sourcemaps = require('gulp-sourcemaps');
var autoprefixer = require('gulp-autoprefixer');
var uglify = require('gulp-uglify');
var concat = require('gulp-concat');
var rename = require('gulp-rename');
var imagemin = require('gulp-imagemin');
var changed = require('gulp-changed');
var del = require('del');
var sequence = require('run-sequence');
var pkg = require('./package.json')

var production = false;

var file = {
  html:   'src/**/*.html',
  scss:   'src/assets/scss/**/*.scss',
  js:     'src/assets/js/src/**/*.js',
}

var page = {
  js:     'src/assets/js/src/page.js',
  scss:   'src/assets/scss/page.scss',
}

var dir = {
  css:    'src/assets/css/',
  js:     'src/assets/js/',
  font:   'src/assets/fonts/',
}

var browsers = [
  'Chrome >= 45',
  'Firefox >= 40',
  'Edge >= 12',
  'Explorer >= 11',
  'iOS >= 9',
  'Safari >= 9',
  'Android 2.3',
  'Android >= 4',
  'Opera >= 30'
];


/*
|--------------------------------------------------------------------------
| Serve
|--------------------------------------------------------------------------
|
*/
gulp.task('reload', function() {
  browserSync.reload();
});

gulp.task('serve', ['sass'], function() {
  browserSync({
    server: 'src/'
  });

  gulp.watch( file.scss, ['sass'] );
  gulp.watch( file.js, function() {
    sequence('js', 'reload');
  });
  gulp.watch( file.html, ['reload'] );
});


/*
|--------------------------------------------------------------------------
| SASS
|--------------------------------------------------------------------------
|
*/
gulp.task('sass', function() {

  var stream = gulp.src(page.scss)
    .pipe( sourcemaps.init() )
    .pipe( rename( { suffix: '.min' } ) )
    .pipe( sass({ importer: tildeImporter, outputStyle: 'compressed' }).on('error', sass.logError) )
    .pipe( autoprefixer({
      browsers: browsers
    }))
    .pipe( sourcemaps.write('.') )
    .pipe( gulp.dest(dir.css) )
    .pipe( browserSync.stream() );

  // Create unminified version if it's in production mode
  if ( production ) {
    stream = gulp.src(page.scss)
      .pipe( sourcemaps.init() )
      .pipe( sass({importer: tildeImporter}).on('error', sass.logError) )
      .pipe( autoprefixer({
        browsers: browsers
      }))
      .pipe( sourcemaps.write('.') )
      .pipe( gulp.dest(dir.css) );
  }

  return stream;

});


/*
|--------------------------------------------------------------------------
| JS
|--------------------------------------------------------------------------
|
*/
gulp.task('js', function(cb) {

  if ( production ) {
    return sequence('js_production_minified', 'js_production_expanded', cb);
  }
  else {
    return gulp.src(page.js)
      .pipe(webpack({
        mode: 'none',
        devtool: 'source-map',
        output: {
          filename: 'page.min.js'
        }
      }))
      .pipe( gulp.dest(dir.js) );
  }

});


gulp.task('js_production_minified', function() {
  return gulp.src(page.js)
    .pipe(webpack({
      mode: 'none',
      devtool: 'source-map',
      output: {
        filename: 'page.min.js'
      },
      plugins: [
        new webpackUglifyJs()
      ]
    }))
    .pipe( gulp.dest(dir.js) );
});


gulp.task('js_production_expanded', function() {
  return gulp.src(page.js)
    .pipe(webpack({
      mode: 'none',
      devtool: 'source-map',
      output: {
        filename: 'page.js'
      }
    }))
    .pipe( gulp.dest(dir.js) );
});


/*
|--------------------------------------------------------------------------
| Copy
|--------------------------------------------------------------------------
|
*/
gulp.task('copyFonts', function() {
  //gulp.src( 'node_modules/@fortawesome/fontawesome-free-webfonts/webfonts/*').pipe(gulp.dest(dir.font));
  gulp.src( 'node_modules/font-awesome/fonts/*').pipe(gulp.dest(dir.font));
  gulp.src( 'node_modules/themify-icons/themify-icons/fonts/*').pipe(gulp.dest(dir.font));
  return gulp.src( 'node_modules/et-line/fonts/*').pipe(gulp.dest(dir.font));
});

gulp.task('distCopy', function() {
  return gulp.src( ['src/**/*', '!src/assets/{js/src,plugin/thesaas,scss}{,/**}'] ).pipe(gulp.dest('dist/'));
});


/*
|--------------------------------------------------------------------------
| Clean /dist directory
|--------------------------------------------------------------------------
|
*/
gulp.task('distClean', function() {
  return del('dist/');
});

/*
|--------------------------------------------------------------------------
| Img
|--------------------------------------------------------------------------
|
*/
gulp.task('img', function() {
  return gulp.src('src/assets/img/**/*.{jpg,jpeg,png,gif}')
    .pipe( imagemin() )
    .pipe( gulp.dest('src/assets/img/') );
});

/*
|--------------------------------------------------------------------------
| Tasks
|--------------------------------------------------------------------------
|
*/
gulp.task('dev', function(cb) {
  sequence('copyFonts', 'sass', 'js', cb);
});

gulp.task('dist', function(cb) {
  production = true;
  sequence('distClean', 'dev', 'distCopy', cb);
});

gulp.task('watch', ['serve']);
gulp.task('default', ['serve']);


v1.5/source/gruntfile.js
/*!
 * TheSaaS's Gruntfile
 * http://thetheme.io/thesaas
 * Copyright 2017 TheThemeio
 */

module.exports = function(grunt) {
  'use strict';

  var autoprefixer = require('autoprefixer')({
    browsers: [
      'Chrome >= 45',
      'Firefox >= 40',
      'Edge >= 12',
      'Explorer >= 11',
      'iOS >= 9',
      'Safari >= 9',
      'Android 2.3',
      'Android >= 4',
      'Opera >= 30'
    ]
  });


  // Project configuration.
  grunt.initConfig({

    // Metadata.
    pkg: grunt.file.readJSON('package.json'),
    banner: '/*!\n' +
            ' * <%= pkg.banner_name %> v<%= pkg.version %> (<%= pkg.homepage %>)\n' +
            ' * Copyright <%= grunt.template.today("yyyy") %> <%= pkg.author %>\n' +
            ' * Licensed under the Themeforest Standard Licenses\n' +
            ' */\n',


    // Task configuration
    // -------------------------------------------------------------------------------


    // Complile SCSS
    //
    sass: {

      app_expanded: {
        options: {
          sourceMap: true,
          outputStyle: 'expanded'
        },
        files: {
          'src/assets/css/<%= pkg.name %>.css': 'src/assets/css/scss/<%= pkg.name %>.scss'
        }
      },

      app_compressed: {
        options: {
          sourceMap: true,
          outputStyle: 'compressed'
        },
        files: {
          'src/assets/css/<%= pkg.name %>.min.css': 'src/assets/css/scss/<%= pkg.name %>.scss'
        }
      },

      /*
      style_expanded: {
        options: {
          sourceMap: true,
          outputStyle: 'expanded'
        },
        files: {
          'src/assets/css/style.css': 'src/assets/css/scss/style.scss'
        }
      },

      style_compressed: {
        options: {
          sourceMap: true,
          outputStyle: 'compressed'
        },
        files: {
          'src/assets/css/style.min.css': 'src/assets/css/scss/style.scss'
        }
      }
      */

    },





    // Watch on SCSS files
    //
    watch: {
      sass_app: {
        files: ['src/assets/css/**/*.scss'],
        tasks: ['sass:app_expanded', 'sass:app_compressed']
      },
      //sass_style: {
      //  files: ['src/assets/css/**/*.scss'],
      //  tasks: ['sass:style_expanded', 'sass:style_compressed']
      //},
      js_app: {
        files: ['src/assets/js/<%= pkg.name %>.js'],
        tasks: ['uglify:app']
      }
      //js_script: {
      //  files: ['src/assets/js/script.js'],
      //  tasks: ['uglify:script']
      //}
    },





    // Browser Sync
    //
    browserSync: {
      dev: {
        bsFiles: {
          src : [
            'src/assets/css/*.min.css',
            'src/assets/js/*.min.js',
            'src/**/*.html'
          ]
        },
        options: {
          watchTask: true,
          server: "src"
        }
      }
    },





    // Clean files and directories
    //
    clean: {
      before_copy: ['dist'],
      after_copy: {
        src: ["dist/theme/assets/css/scss",
              "dist/theme/assets/vendor",
              "dist/theme/assets/img/**/*",
              "dist/source/src/assets/img/**/*"
              ],
      }
    },





    // Copy files
    //
    copy: {
      dist: {
        files: [
          {expand: true, cwd: 'src/', src: ['**'], dest: 'dist/theme'},
          {expand: true, cwd: '.',    src: ['demo.html', 'support.html', 'CHANGELOG.md'], dest: 'dist'}
        ],
      },

      source: {
        files: [
          {expand: true, cwd: 'src/', src: ['**'],                           dest: 'dist/source/src'},
          {expand: true, cwd: '.',    src: ['package.json', 'gruntfile.js'], dest: 'dist/source'}
        ]
      },

      placeholder: {
        files: [
          {expand: true, cwd: 'placeholders/', src: ['**'], dest: 'dist/theme/assets/img'},
          {expand: true, cwd: 'placeholders/', src: ['**'], dest: 'dist/source/src/assets/img'}
        ],
      },

      dev: {
        files: [
          {expand: true, cwd: 'src/assets/vendor/font-awesome/fonts',  src: ['**'], dest: 'src/assets/fonts/'},
          {expand: true, cwd: 'src/assets/vendor/themify-icons/fonts', src: ['**'], dest: 'src/assets/fonts/'},
          {expand: true, cwd: 'src/assets/vendor/et-line-font/fonts', src: ['**'], dest: 'src/assets/fonts/'},
        ]
      }
    },





    // Concat plugins to make core.min
    //
    concat: {

      core: {
        files: {
          // Javascript
          'src/assets/js/core.min.js':
          [
            'src/assets/vendor/jquery.min.js',
            'src/assets/vendor/tether.min.js',
            'src/assets/vendor/bootstrap/js/bootstrap.min.js',
            'src/assets/vendor/smoothscroll.min.js',
            'src/assets/vendor/parallax/parallax.min.js',
            'src/assets/vendor/aos/aos.js',
            'src/assets/vendor/lity/lity.min.js',
            'src/assets/vendor/swiper/js/swiper.jquery.min.js',
            'src/assets/vendor/imagesloaded.pkgd.min.js', // For shuffle.js
            'src/assets/vendor/shuffle/shuffle.min.js',
            'src/assets/vendor/jquery.waypoints.min.js', // For countTo
            'src/assets/vendor/jquery.countTo/jquery.countTo.min.js',
            'src/assets/vendor/jquery.countdown/jquery.countdown.min.js',
            'src/assets/vendor/typed/typed.min.js',
            'src/assets/vendor/constellation.min.js',
            'src/assets/vendor/object-fit/objectFitPolyfill.min.js',
          ],

          // CSS
          'src/assets/css/core.min.css':
          [
            'src/assets/vendor/bootstrap/css/bootstrap.min.css',
            'src/assets/vendor/font-awesome/css/font-awesome.min.css',
            'src/assets/vendor/themify-icons/css/themify-icons.css',
            'src/assets/vendor/et-line-font/css/style.min.css',
            'src/assets/vendor/aos/aos.css',
            'src/assets/vendor/lity/lity.min.css',
            'src/assets/vendor/swiper/css/swiper.min.css',
          ]
        }
      }

    },






    // Uglify JS files
    //
    uglify: {
      options: {
        mangle: true,
        preserveComments: /^!|@preserve|@license|@cc_on/i,
        banner: '<%= banner %>'
      },
      app: {
        src:  'src/assets/js/<%= pkg.name %>.js',
        dest: 'src/assets/js/<%= pkg.name %>.min.js'
      },
      //script: {
      //  src:  'src/assets/js/script.js',
      //  dest: 'src/assets/js/script.min.js'
      //}
      //core: {
      //  src:  'src/assets/js/core.min.js',
      //  dest: 'src/assets/js/core.min.js'
      //}
    },





    // CSS build configuration
    //
    scsslint: {
      options: {
        bundleExec: true,
        config: 'src/assets/css/scss/.scss-lint.yml',
        reporterOutput: null
      },
      core: {
        src: ['src/assets/css/scss/*.scss']
      }
    },





    // Do some post processing on CSS files
    //
    postcss: {
      options: {
        processors: [
          autoprefixer,
          require('postcss-flexbugs-fixes')
        ]
      },
      expanded: {
        src: 'src/assets/css/<%= pkg.name %>.css'
      },
      compressed: {
        src: 'src/assets/css/<%= pkg.name %>.min.css'
      }
    },





    // Minify CSS files
    //
    cssmin: {
      options: {
        compatibility: 'ie9',
        keepSpecialComments: '*',
        sourceMap: false,
        advanced:  false
      },
      plugins: {
        src:  'src/assets/css/core.min.css',
        dest: 'src/assets/css/core.min.css'
      }
    }

    // -------------------------------------------------------------------------------
    // END Task configuration

  });


  // These plugins provide necessary tasks.
  require('load-grunt-tasks')(grunt, { scope: 'devDependencies', pattern: ['grunt-*'] });
  require('autoprefixer')(grunt);
  //require('time-grunt')(grunt);

  // Run "grunt" to watch SCSS and JS files as well as running browser-sync
  grunt.registerTask('default', ['browserSync', 'watch']);
  grunt.registerTask('serve', ['browserSync', 'watch']);

  // Run "grunt dist" to publish the template in a ./dist folder
  grunt.registerTask('dist',
    [
      'clean:before_copy',
      'dev',
      'copy:dist',
      'copy:source',
      'clean:after_copy',
      'copy:placeholder'
    ]
  );

  // Run "grunt dev" to make sure your CSS and JS files are up to date for development
  grunt.registerTask('dev',
    [
      'sass',
      'copy:dev',
      'concat',
      //'replace',
      'uglify',
      'cssmin',
      'postcss'
    ]
  );


};
